app/lib/feature_flags.rb

```diff
@@ -3,21 +3,28 @@ class FeatureFlags
     example_2125_flag: { from: "2125-01-01", public: false },
   }.freeze
 
-  STATIC_FLAGS = %i[sentry cw_forms special_applicant_groups self_employed].freeze
+  # the values of some feature flags will come from the session and not the env variables.
+  # "global" - feature flag value should be derived from the env variable
+  # "session" - feature flag value should be derived from the session_data of the check
+  STATIC_FLAGS = { sentry: "global", cw_forms: "global", special_applicant_groups: "session", self_employed: "session" }.freeze
 
   class << self
-    def enabled?(flag)
+    def enabled?(flag, session_data = nil)
+      if session_data && session_data["feature_flags"]&.key?(flag.to_s)
```

Comment:
I wonder if we should have the same checking the key is a recognised one here as we do for global ones?

Reply:
I might not be thinking this through correctly, but I'm not sure how that would work - the 'flag unrecognised' fallback is still the final fallback in this code. If the flag isn't found in the `session_data`, the code will fall back to the `STATIC_FLAGS`, then the `ENABLED_AFTER_DATE` flags and then finally will fall back to the error if the flag can't be found in any of those places. But I might be misunderstanding!

Reply:
So I think there's two different things:

One is where the code asks for a flag, and it's not in session data, so it falls back as you say and we get the goodness like error handling and defaults.

The other was that I saw `if STATIC_FLAGS.key?(flag)` and interpreted it as us wanting to only look for allow-listed flag names in the env, and therefore figured we'd want to only look in the session for allow-listed feature flag names. But now looking at it more I think maybe that's not the reasoning at all, so it's fine.